src/main/java/coresearch/cvurl/io/request/CompressedStringBodyHandler.java

```diff
@@ -0,0 +1,44 @@
+package coresearch.cvurl.io.request;
+
+import coresearch.cvurl.io.constant.HttpContentEncoding;
+import coresearch.cvurl.io.constant.HttpHeader;
+import coresearch.cvurl.io.exception.ResponseBodyHandlingException;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.http.HttpResponse;
+import java.util.Optional;
+import java.util.zip.GZIPInputStream;
+
+import static coresearch.cvurl.io.internal.utils.ResponseInfoUtils.getEncoding;
+
+public class CompressedStringBodyHandler implements HttpResponse.BodyHandler<String> {
+
+    CompressedStringBodyHandler() {
```

Comment:
Why do you need it?

Reply:
We need it to create get HttpResponse<String> where string is created from decompressed response body.

Reply:
I am asking about constructor

Reply:
this to make it package private to allow to use it only within request package

Reply:
In this case, make sense to do the same for CompressedBodyHandler

Reply:
done